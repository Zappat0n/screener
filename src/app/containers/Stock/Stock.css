.stock {
  font: normal normal 400 1rem/1rem 'Open Sans', sans-serif;
}


.stock .price,
.stock .change {
  text-align: right;
}

