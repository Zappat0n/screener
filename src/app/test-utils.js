import React from 'react';
import { createStore, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import indexReducer from './reducers/indexSlice';
import stockReducer from './reducers/stockSlice';
import profileReducer from './reducers/profileSlice';
import filterReducer from './reducers/filterSlice';

const rootReducer = combineReducers({
  indexes: indexReducer, stocks: stockReducer, profile: profileReducer, filter: filterReducer,
});

const initialState = {
  stocks: {
    AAPL: {
      price: 100,
      companyName: 'Apple Inc',
      changes: 1,
      sector: 'Technology',
    },
  },
  profile: {
    data: {
      image: '',
      companyName: 'Microsoft',
      price: 100,
      changes: 1,
      mktCap: 1000000,
      range: '0-100',
      country: 'US',
    },
  },
};

const renderComponent = (component) => render(
  <Provider store={createStore(rootReducer, initialState)}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export * from '@testing-library/react';
export { renderComponent };
