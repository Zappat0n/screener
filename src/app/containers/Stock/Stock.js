import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import './Stock.css';

const Stock = (props) => {
  const { ticker, handleClick } = props;
  const data = useSelector((state) => (state.stocks[ticker] ? state.stocks[ticker] : ''));

  const getPercentage = (change, price) => {
    if (price !== '') {
      return (change !== 0 ? `${(Math.round((change * 10000) / price) / 100)}%` : '0%');
    }
    return '';
  };

  const getColor = () => (data && data.changes < 0 ? 'colorRed' : 'colorBlue');

  return (
    <tr className="stock">
      <td>
        {data.companyName}
      </td>
      <td className={`price ${getColor()}`}>
        {data.price}
      </td>
      <td className={`change ${getColor()}`}>
        {getPercentage(data.changes, data.price)}
      </td>
      <td>
        {data.industry}
      </td>
      <td>
        <button type="button" onClick={(event) => handleClick(event, ticker)}>*</button>
      </td>
    </tr>
  );
};

Stock.propTypes = {
  ticker: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Stock;
