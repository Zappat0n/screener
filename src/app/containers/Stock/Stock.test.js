import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import Stock from './Stock';
import { renderComponent, rendererForComponent } from '../../test-utils';

test('Renders Stock correctly', () => {
  renderComponent(
    <table>
      <tbody>
        <Stock ticker="AAPL" />
      </tbody>
    </table>,
  );
  expect(screen.getByText('Apple Inc')).toBeInTheDocument();
  expect(screen.getAllByText('100')[0]).toBeInTheDocument();
  expect(screen.getByText('1%')).toBeInTheDocument();
  expect(screen.getByText('Technology')).toBeInTheDocument();
});

test('Stock Match SnapShot', () => {
  const tree = rendererForComponent(
    <table>
      <tbody>
        <Stock ticker="AAPL" />
      </tbody>
    </table>,
  );
  expect(tree).toMatchSnapshot();
});
