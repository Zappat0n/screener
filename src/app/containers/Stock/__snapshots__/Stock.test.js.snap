// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock Match SnapShot 1`] = `
<table>
  <tbody>
    <tr
      className="stock"
    >
      <td>
        <a
          className="link "
          href="/stock/AAPL"
          onClick={[Function]}
        >
          Apple Inc
        </a>
      </td>
      <td>
        <a
          className="link "
          href="/stock/AAPL"
          onClick={[Function]}
        >
          100
        </a>
      </td>
      <td>
        <a
          className="link colorBlue"
          href="/stock/AAPL"
          onClick={[Function]}
        >
          1%
        </a>
      </td>
      <td>
        <a
          className="link "
          href="/stock/AAPL"
          onClick={[Function]}
        >
          Technology
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
